src/bin/main.rs

```diff
@@ -305,50 +298,108 @@ fn main() {
         };
     }
 
+    match (&paths.code_files, &paths.code_paths) {
+        (Some(code_files), Some(code_paths)) if code_files.len() != code_paths.len() => {
+            eprintln!(
+                "If argument code_paths is given in the toml file, it must have as many elements as argument code_files",
+            );
+            return;
+        }
+        _ => (),
+    }
+
+    let mut track_copy_dest = HashMap::new();
     if let Some(code_dir) = code_dir {
-        if let Some(code_files) = code_files {
-            for code_file in &code_files {
-                eprintln!("Copying code file {:?}", code_file);
-                let mut file_path = code_dir.clone();
-                file_path.push(code_file);
-                fs::create_dir_all(file_path.parent().unwrap()).unwrap();
-                match fs::copy(&code_file, &file_path) {
-                    Ok(_) => {}
-                    Err(err) => eprintln!("  --> Error copying code file {:?}: {}", code_file, err),
+        if let Some(code_file_patterns) = paths.code_files {
+            for (idx, code_file_pattern) in code_file_patterns.iter().enumerate() {
+                let code_path = paths.code_paths.as_ref().map(|code_paths| &code_paths[idx]);
+                for code_file in PathUtil::list_files(code_file_pattern).unwrap() {
+                    let out_path = code_path.map_or(code_file.clone(), |code_path| {
+                        modify_path(&code_file, &code_path)
+                    });
+                    match track_copy_dest.entry(out_path.clone()) {
+                        Occupied(entry) => {
+                            eprintln!(
+                                "ERROR: Attempted to copy multiple code files to {:?}: from {:?} and {:?}",
+                                out_path, entry.get(), code_file
+                            );
+                            return;
+                        }
+                        Vacant(entry) => {
+                            entry.insert(code_file.clone());
+                        }
+                    }
+                    let mut file_path = code_dir.clone();
+                    file_path.push(out_path.clone());
+
+                    eprintln!("Copying code file {:?} to {:?}", code_file, out_path);
```

Comment:
I just noticed that you either use `.to_str().unwrap()` or `{:?}` everywhere to print `Path`s. There is [`Path::display`](https://doc.rust-lang.org/stable/std/path/struct.Path.html#method.display) specifically for that purpose.